src/controller/backend/Endpoints/UserManagementEndpoints.cs

```diff
@@ -16,156 +10,99 @@ namespace Backend.Endpoints;
-
-            var result = await userManager.SetUserNameAsync(user, model.Username);
-
-            return result.Succeeded ? Results.Ok() : Results.BadRequest(result.Errors);
-
-        }).RequirePermission(UserClaims.ProfileChangeUsername);
+            var userEntity = await context.Users
+                .Include(s => s.UserProfile).FirstOrDefaultAsync(s => s.UserName == user);
+            if (userEntity is null)
+                return Results.NotFound("User not found");
+            if (!userEntity.CanChangeUsername)
+                return Results.Forbid();
+            userEntity.UserName = model.Username;
+            var result = await context.SaveChangesAsync();
+            return result == 1 ? Results.Ok() : Results.InternalServerError();
+        }).RequirePermission(UserClaims.UserChangeUserUsername);
 
-        group.MapPost("/password", async (
-            HttpContext context,
+        group.MapPost("/users/{user}/password", async (
             UserManager<SystemUser> userManager,
-            [FromBody] PasswordModel model) =>
+            [FromRoute] string user,
+            [FromBody] ChangePasswordModel model) =>
         {
-            var user = userManager.GetUserAsync(context.User).Result;
-            if (user is null)
-                return Results.Unauthorized();
-
-            var result = await userManager.ChangePasswordAsync(user, model.CurrentPassword, model.NewPassword);
-
+            var userEntity = await userManager.FindByNameAsync(user);
+            if (userEntity is null)
+                return Results.NotFound("User not found");
+            await userManager.RemovePasswordAsync(userEntity);
+            var result = await userManager.AddPasswordAsync(userEntity, model.NewPassword);
             return result == IdentityResult.Success ? Results.Ok() : Results.BadRequest(result.Errors);
+        }).RequirePermission(UserClaims.UserChangeUserPassword);
 
-        }).RequirePermission(UserClaims.ProfileChangePassword);
-
-        group.MapGet("/backup", () =>
+        group.MapPost("/users/{user}/isactive", async (
+            [FromRoute] string user,
+            [FromBody] ChangeIsActiveModel model,
+            SystemDbContext context) =>
         {
-            // TODO: Implement backup functionality
-        }).RequireAuthorization("token");
-
-        group.MapGet("/language", async (
-            HttpContext context,
-            UserManager<SystemUser> userManager,
+            var userEntity = context.Users.FirstOrDefault(s => s.UserName == user);
+            if (userEntity is null)
+                return Results.NotFound();
+            if(userEntity.UserName == "admin")
+                return Results.Forbid();
+            userEntity.IsActive = model.IsActive;
+            var result = await context.SaveChangesAsync();
+            return result == 1 ? Results.Ok() : Results.InternalServerError();
+        }).RequirePermission(UserClaims.UserChangeUserUsername);
```

Comment:
IsActive endpoint with UserClaims.UserChangeUserUsername permissions? I haven't checked out all our different permissions but this seems off. Also, is this supposed to be a admin endpoint? Maybe we can mark admin endpoints as like admin only or something like that? Or did you intend to also split up all the admin functions into separate permissions because we really don't need to. Admin settings are for admin only. Period.

Reply:
it's supposed to be like this, trust :3